src/hdf5_summary.F90

```diff
@@ -27,48 +25,47 @@ module hdf5_summary
 
   subroutine hdf5_write_summary()
 
-    character(MAX_FILE_LEN) :: filename = "summary.h5"
+    integer(HID_T) :: file_id
 
     ! Create a new file using default properties.
-    call su % file_create(filename)
+    file_id = file_create("summary.h5")
 
     ! Write header information
-    call hdf5_write_header()
-
-    ! Write eigenvalue information
-    if (run_mode == MODE_EIGENVALUE) then
+    call hdf5_write_header(file_id)
 
-      ! Write number of particles
-      call su % write_data(n_particles, "n_particles")
+    ! Write number of particles
+    call write_dataset(file_id, "n_particles", n_particles)
+    call write_dataset(file_id, "n_batches", n_batches)
+    call write_attribute_string(file_id, "n_particles", &
+         "description", "Number of particles per generation")
+    call write_attribute_string(file_id, "n_batches", &
+         "description", "Total number of batches")
 
-      ! Use H5LT interface to write n_batches, n_inactive, and n_active
-      call su % write_data(n_batches, "n_batches")
-      call su % write_data(n_inactive, "n_inactive")
-      call su % write_data(n_active, "n_active")
-      call su % write_data(gen_per_batch, "gen_per_batch")
+      ! Write eigenvalue information
+    if (run_mode == MODE_EIGENVALUE) then
+      ! write number of inactive/active batches and generations/batch
+      call write_dataset(file_id, "n_inactive", n_inactive)
+      call write_dataset(file_id, "n_active", n_active)
+      call write_dataset(file_id, "gen_per_batch", gen_per_batch)
 
       ! Add description of each variable
-      call su % write_attribute_string("n_particles", &
-           "description", "Number of particles per generation")
-      call su % write_attribute_string("n_batches", &
-           "description", "Total number of batches")
-      call su % write_attribute_string("n_inactive", &
+      call write_attribute_string(file_id, "n_inactive", &
            "description", "Number of inactive batches")
-      call su % write_attribute_string("n_active", &
+      call write_attribute_string(file_id, "n_active", &
            "description", "Number of active batches")
-      call su % write_attribute_string("gen_per_batch", &
+      call write_attribute_string(file_id, "gen_per_batch", &
            "description", "Number of generations per batch")
     end if
 
-    call hdf5_write_geometry()
-    call hdf5_write_materials()
-    call hdf5_write_nuclides()
+    call hdf5_write_geometry(file_id)
```

Comment:
Just curious, since we are using all HDF5 now, any reason to explicitly (and redundantly) include "hdf5_" in these subroutine signatures?